nixos/tests/openbao.nix

```diff
@@ -0,0 +1,108 @@
+{ lib, ... }:
+let
+  certs = import ./common/acme/server/snakeoil-certs.nix;
+  domain = certs.domain;
+in
+{
+  name = "openbao";
+
+  meta.maintainers = with lib.maintainers; [ kranzes ];
+
+  nodes.machine =
+    { config, ... }:
+    {
+      security.pki.certificateFiles = [ certs.ca.cert ];
+
+      networking.extraHosts = ''
+        127.0.0.1 ${domain}
+      '';
+
+      services.openbao = {
+        enable = true;
+
+        settings = {
+          ui = true;
+
+          listener = {
+            default = {
+              type = "tcp";
+              tls_cert_file = certs.${domain}.cert;
+              tls_key_file = certs.${domain}.key;
+            };
+
+            unix = {
+              type = "unix";
+            };
+          };
+
+          cluster_addr = "https://127.0.0.1:8201";
+          api_addr = "https://${domain}:8200";
+
+          storage.raft = {
+            path = "/var/lib/openbao";
+            node_id = config.networking.hostName;
```

Comment:
Should we set this as a default?

Reply:
Not sure, probably not

Reply:
I don't think we should set a default when there's many other storage backends. I think users should make a choice first.

Reply:
OpenBao sets a default if none is provided; did you run into conflicts using the default?

Reply:
I didn't know there was a default value, where is that documented?

Reply:
Ah, I see, it isn't documented in https://openbao.org/docs/configuration/storage/raft/#raft-parameters 

The actual auto-generation fact is mentioned here: https://openbao.org/docs/concepts/integrated-storage/#manual-recovery-using-peersjson

> This can be found in the config file, or inside the node-id file in the server's data directory if it was auto-generated.

I'll open a PR to update the first link to include that fact.